### DISCUSSION TITLE
How can I show more of stack string in the empty space?

### DISCUSSION THREAD

**Discussion by rew1nter** [hugsy/gef#1074] [Q&A] [answered]
Is there a way to print more instead of `[...]`
![image](https://github.com/hugsy/gef/assets/64508791/b8a90834-4e76-4e3c-9a7c-83c51fa5e21c)

**Comment by hugsy** [comment]
GEF hints you that it's an ascii string, so you can use the normal GDB command `x/s $addr`

**Comment by rew1nter** [comment]
yeah that's how I've been doing it. but it is easier to see the full ASCII string printed in UI rather than typing the command every time. Iirc peda prints the full ascii text

**Reply by hugsy** [reply, marked as answer]
Well this value is [hardcoded on purpose](https://github.com/hugsy/gef/blob/main/gef.py#L131) because having unterminated strings can get quickly out of hands from a UI and performance perspective .

So I guess, you have 2 options:
- edit `GEF_MAX_STRING_LENGTH` to a bigger value that you find suitable
- directly read the memory from [your own custom context pane](https://hugsy.github.io/gef/api/#customizing-context-panes) using `gef.memory.read_cstring($addr, max_length=$whatever_you_want)`

Option 1 is faster, but more quick'n dirty.
Option 2 is the more suitable IMO since it's a personal UI preference

**Comment by rew1nter** [comment]
Ummm could you please make sure changing GEF_MAX_STRING_LENGTH has an effect on your end? 
I just changed the value and it seems to have no effect. Still only 50 Ascii chars are being printed

**Reply by hugsy** [reply]
Great catch, this was a bug. @Grazfather fixed it, you should be good to go after pulling the latest version.

**Comment by Grazfather** [comment]
Seems we hardcode it in `def dereference_from` around line 8507 to 50 characters. We could parameterize that, and in the meantime you could just bump that.

**Reply by Grazfather** [reply]
This is fixed on main now.

**Comment by rew1nter** [comment]
btw setting this variable to a large value like 200 shows no ASCII text whatsoever. It probably breaks the UI. You guys could add a check or comment for the largest possible value, better yet, add an option for maximum length
![image](https://github.com/hugsy/gef/assets/64508791/81ea67d6-8610-4d64-a32a-152ff26e39e9)

**Comment by Grazfather** [comment]
That's maybe why it's hardcoded. While we are hardcoding it, we have no reason to check for this sort of thing.
